spmd/tensor/parallel/style.py

```diff
@@ -27,6 +21,27 @@ class ParallelStyle(ABC):
     _prepare_input: Optional[_Prepare_Input_Func_Type]
     _prepare_output: Optional[_Prepare_Output_Func_Type]
 
+    @abstractmethod
+    def __init__(self, _prepare_input, _prepare_output) -> None:
+        self._prepare_input = _prepare_input
+        self._prepare_output = _prepare_output
+
+
+class PairwiseParallel(ParallelStyle):
+    """
+    PairwiseParallel concatenate colwise and rowwise styles as a fixed
+    pair like what Megatron-LM(https://arxiv.org/abs/1909.08053) is doing.
+    We assume both input and output needs to a replicate DTensor.
+
+    .. warning::
+        We need to the module to be distributed as a
```

Comment:
looks like the some typos, i'm wondering if this would make it more clear: "PairwiseParallel only supports ``nn.Multihead Attention``, ``nn.Transformer`` or MLP Layers (i.e. even number ``nn.Linear`` layers) for now"

Reply:
Changed the comment here. But I would not use the world MLP since L means random layers of perceptrons.